webclient/webclient/src/main/java/io/helidon/webclient/RequestConfiguration.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.webclient;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import io.helidon.media.common.MessageBodyReader;
+import io.helidon.media.common.MessageBodyWriter;
+import io.helidon.webclient.spi.ClientService;
+
+/**
+ * Configuration of specific request.
+ */
+class RequestConfiguration extends ClientConfiguration {
+
+    private final URI requestURI;
+    private final ClientServiceRequest clientServiceRequest;
+    private final List<ClientService> services;
+    private final Set<MessageBodyReader<?>> requestReaders;
+    private final Set<MessageBodyWriter<?>> requestWriters;
+
+    private RequestConfiguration(Builder builder) {
+        super(builder);
+        requestURI = builder.requestURI;
+        clientServiceRequest = builder.clientServiceRequest;
+        services = builder.services;
+        requestReaders = builder.requestReaders;
+        requestWriters = builder.messageBodyWriters;
+    }
+
+    URI requestURI() {
+        return requestURI;
+    }
+
+    ClientServiceRequest clientServiceRequest() {
+        return clientServiceRequest;
+    }
+
+    List<ClientService> services() {
+        return services;
+    }
+
+    Set<MessageBodyReader<?>> requestReaders() {
```

Comment:
Thinking-out-loud.

I don't know if having a configuration for request/response specific readers/writers is really useful.
Instead you could expose the context from request/response and let the user add these manually for each request/response.

If we think there is a need to have HTTP client specific readers/writers for all client request/response, then maybe we should consider updating `MediaSupport`, `MessageBodyReaderContext` and `MessageBodyWriterContext` to support creating a parented `MediaSupport`.

This way you would only need to pass-in a `MediaSupport` instance to `RequestConfiguration`.

Reply:
Ah this is great you are mentioning this. I have a note to create issue for that. I think that having builder on `MediaSupport` which allows you to insert explicit parent, would be great for my usecase. I didn't added yet, because I wanted to talk to you about it before.

In case of the exposing of context, I do not think that is what we should do. 

The reason why I opted for specific readers/writers for each request/response is, that it allows you to have control over entity handling. Basically you can have two or more ways how to handle specific entity type in your code and this helps you to achieve this without interfering with other request where it is not desirable.

Reply:
There should be a parented context dedicated to each request, that would be symmetrical to how it's done in the webserver currently. We can talk more about this when you have some time.